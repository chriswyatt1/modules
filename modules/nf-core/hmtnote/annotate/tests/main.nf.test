
nextflow_process {

    name "Test Process HMTNOTE_ANNOTATE"
    script "../main.nf"
    process "HMTNOTE_ANNOTATE"
    config "./nextflow.config"

    tag "modules"
    tag "modules_nfcore"
    tag "hmtnote"
    tag "hmtnote/annotate"

    test("test-hmtnote-annotate") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
				    file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/vcf/test.vcf', checkIfExists: true)
				]

                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
					path(process.out.vcf[0][1]).vcf.variantsMD5,
					process.out.versions
					).match()
				}
            )
        }
    }

    test("test-hmtnote-annotate-stub") {
        options '-stub'
        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
				    file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/vcf/test.vcf', checkIfExists: true)
				]

                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

}
