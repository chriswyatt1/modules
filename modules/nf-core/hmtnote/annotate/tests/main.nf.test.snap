{
    "test-hmtnote-annotate": {
        "content": [
            "98d1056442ec88659f26dfb1251b6f74",
            [
                "versions.yml:md5,8ee58af72515ea6d362a7721c3ba1340"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-02T13:58:45.987177"
    },
    "test-hmtnote-annotate-stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_annotated.vcf:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,8ee58af72515ea6d362a7721c3ba1340"
                ],
                "vcf": [
                    [
                        {
                            "id": "test"
                        },
                        "test_annotated.vcf:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,8ee58af72515ea6d362a7721c3ba1340"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-02T13:58:54.748392"
    }
}